code/game/machinery/computer/depot.dm

```diff
@@ -164,12 +201,19 @@
 	icon_screen = "explosive"
 	req_access = list()
 	alerts_when_broken = TRUE
-
-/obj/machinery/computer/syndicate_depot/selfdestruct/get_menu(mob/user)
-	var/menutext = {"<B>Syndicate Depot Fusion Reactor Control</B><HR>
-	<BR><BR><a href='?src=[UID()];primary=1'>Disable Containment Field</a>
-	<BR>"}
-	return menutext
+	window_height = 200 // this might appear big, but it has to have space for the lockout alert
+
+/obj/machinery/computer/syndicate_depot/selfdestruct/ui_data(mob/user)
+	var/list/data = ..()
+	data["rows"].Add(list(list(
+		"title" = "Reactor Containment Fields",
+		"status" = (istype(depotarea) && !depotarea.used_self_destruct) ? "Online" : "Offline",
+		"buttontitle" = (istype(depotarea) && !depotarea.used_self_destruct) ? "Disable" : null,
```

Comment:
Question, how does this button render if the title is `null`?

Reply:
It doesn't. Per above, 'null' means don't render a button.